packages/block-editor/src/components/iframe/index.js

```diff
@@ -264,7 +271,7 @@ function Iframe( {
 	// mode. Also preload the styles to avoid a flash of unstyled
 	// content.
 	const html = `<!doctype html>
-<html>
+<html lang="${ siteLocale?.lang }" dir="${ siteLocale?.isRTL ? 'rtl' : 'ltr' }">
```

Comment:
Here's another idea: what if the initial HTML as a whole is a setting? This all of this stuff is customisable from outside the iframe component.